src/components/Tokens/TokenDetails/InvalidTokenDetails.tsx

```diff
@@ -23,32 +29,56 @@ const InvalidDetailsText = styled.span`
   line-height: 28px;
 `
 
-const TokenExploreButton = styled.button`
-  border: none;
-  border-radius: 12px;
-  background-color: ${({ theme }) => theme.accentAction};
+const TokenExploreButton = styled(ButtonPrimary)`
+  width: fit-content;
   padding: 12px 16px;
+  border-radius: 12px;
 
   color: ${({ theme }) => theme.textPrimary};
   font-size: 16px;
   font-weight: 600;
 `
 
-export default function InvalidTokenDetails({ chainName }: { chainName?: string }) {
+export default function InvalidTokenDetails({
+  pageChainId,
+  isInvalidAddress,
+}: {
+  pageChainId: SupportedChainId
+  isInvalidAddress?: boolean
+}) {
+  const { chainId } = useWeb3React()
   const navigate = useNavigate()
+  const selectChain = useSelectChain()
+
+  // if the token's address is valid and the chains match, it's a non-existant token
+  const isNonExistantToken = !isInvalidAddress && pageChainId === chainId
```

Comment:
existant not a word 😅 
```suggestion
  const isNonExistentToken = !isInvalidAddress && pageChainId === chainId
```